src/entries/popup/handlers/wallet.ts

```diff
@@ -157,13 +158,33 @@ export const sendTransaction = async (
         throw new Error('Unsupported hardware wallet');
     }
   } else {
-    return walletAction(
+    const transactionResponse = await walletAction<TransactionResponse>(
       'send_transaction',
       params,
-    ) as unknown as TransactionResponse;
+    );
+
+    return deserializeBigNumbers(transactionResponse);
   }
 };
 
+function deserializeBigNumbers<T>(obj: T) {
```

Comment:
not 100% sure about this solution, open to suggestions
but `walletAction` bridges to the background worker and back, in this back and forth it serializes BigNumbers as `{ type: 'BigNumber', _hex: "0x..." }`
so sometimes we got an `[object Object]` instead of the bignumber in some places

Reply:
I started to do it and got afraid of changing the type and breaking in unexpected places, because it's being casted as `TransactionResponse` and that is why we didn't caught it for so long to begin with
to do it without changing the type, I'd have to then deserialize by hand like 
```ts
return {
 ...response,
 maxBaseFee: BigNumber.from(response.maxBaseFee)
 ...
}
```
which is more error prone and effort

another idea is to do this generic BigNumber deserialization inside `walletAction` so we don't have to ever think about it again in other places

Reply:
okok sounds good, can we move this to a util file somewhere else?

Reply:
+1